iwyu_tool.py

```diff
@@ -298,6 +298,11 @@ def start(self, verbose):
 def fixup_compilation_db(compilation_db):
     """ Canonicalize paths in JSON compilation database. """
     for entry in compilation_db:
+        # Convert relative paths to absolute ones if possible, based on the entry's directory.
+        if 'directory' in entry and not os.path.isabs(entry['file']):
+            if os.path.exists(entry['directory']):
```

Comment:
Hm, this seems speculative. If the directory doesn't exist, why is it in the compilation database?

I would expect something like:
```
filename = entry['file']
dirpath = entry.get('directory')
if dirpath and not os.path.isabs(filename):
    filename = os.path.join(dirpath, filename)

entry['file'] = os.path.realpath(filename)
```

And I think you need to adjust my cwd test to not have a `directory` in the entry.